### ISSUE TITLE
Support for named and abbreviated timezones in format() (e.g. "Australian Eastern Daylight Time" / "AEDT")

### ISSUE THREAD

**Issue by philipwindeyer** [js-fns/date-fns#3943]
We recently upgraded to date-fns v4.1.0 (from 3.6.0), with the added intention of replacing our use of `formatInTimeZone()` from `date-fns-tz` with `tz` from `@date-fns/tz` and `format()`.

I.e. from `formatInTimeZone(dateObject, 'Australia/Sydney', dateFormat, { locale: enAU })` to `format(dateObject, dateFormat, { in: tz('Australia/Sydney'), locale: enAU })`.

We use the pattern `E d MMM, yyyy - HH:mm zzz`, and expect output like `Thu 31 Mar, 2022 - 11:39 AEDT` (noting "AEDT" specifically), however this formatting option appears to not yet be supported by `format()`.

May we request that this formatting option (along with named timezone support e.g. "Australian Eastern Daylight Time") be added please? Or when this feature might be added?

During testing/experimenting I did notice that the named timezone is included in the output of `toString()` on a `TZDate` object.
I also acknowledge note 6 in the docs for [format()](https://date-fns.org/v4.1.0/docs/format), i.e. "Specific non-location timezones are currently unavailable in date-fns, so right now these tokens fall back to GMT timezones"

**Comment by rubendiazjs** [comment]
Exact same issue here. Did you find any workaround so far @philipwindeyer?

**Comment by soren121** [comment]
Duplicate of #873, but certainly more important now that date-fns advertises first-class timezone support. Seems like this was missed when implementing timezones.

**Comment by stephent** [comment]
Agreed - this is important. I'm finding that `formatInTimeZone` doesn't give the expected output when the peer dependency version of `date-fns` is 4.x, as the call is simply an adjusted date from `toZonedTime` passed through the `date-fns` itself.

**Comment by imaksp** [comment]
faced same issue today, for now thinking omitting zone from format, and adding it manually using string concat.
